Add CHANGELOG for v5.28.0 (#18011)
hashicorp/terraform-provider-google: CHANGELOG.md

## 5.28.0 (Unreleased)

DEPRECATIONS:
* integrations: deprecated `create_sample_workflows` and `provision_gmek` fields in `google_integrations_client`.  ([#17945](https://github.com/hashicorp/terraform-provider-google/pull/17945))

FEATURES:
* **New Data Source:** `google_storage_buckets` ([#17960](https://github.com/hashicorp/terraform-provider-google/pull/17960))
* **New Resource:** `google_compute_security_policy_rule` ([#17937](https://github.com/hashicorp/terraform-provider-google/pull/17937))

> * **New Resource:** `google_privileged_access_manager_entitlement` (beta) ([#17940](https://github.com/hashicorp/terraform-provider-google/pull/17940))

Reviewer: ```suggestion
```

beta-only

IMPROVEMENTS:
* alloydb: added `maintenance_update_policy` field to `google_alloydb_cluster` resource ([#17954](https://github.com/hashicorp/terraform-provider-google/pull/17954))

> * bigquery: promoted `external_dataset_reference` in `bigquery_dataset` to GA ([#17944](https://github.com/hashicorp/terraform-provider-google/pull/17944))

Reviewer 2: ```suggestion
* bigquery: promoted `external_dataset_reference` in `google_bigquery_dataset` to GA ([#17944](https://github.com/hashicorp/terraform-provider-google/pull/17944))
```

Author: Thanks!

> * composer: promoted `config.0.software_config.0.image_version` in-place updates to GA in resource `google_composer_environment` ([#17986](https://github.com/hashicorp/terraform-provider-google/pull/17986))

Reviewer 2: ```suggestion
* composer: promoted `config.software_config.image_version` in-place update to GA in resource `google_composer_environment` ([#17986](https://github.com/hashicorp/terraform-provider-google/pull/17986))
```

* compute: added `server_tls_policy` field to `google_compute_region_target_https_proxy` resource ([#17934](https://github.com/hashicorp/terraform-provider-google/pull/17934))
* logging: added `intercept_children` field to `google_logging_organization_sink` and `google_logging_folder_sink` resources ([#17932](https://github.com/hashicorp/terraform-provider-google/pull/17932))
* monitoring: added `service_agent_authentication` field to `google_monitoring_uptime_check_config` resource ([#17929](https://github.com/hashicorp/terraform-provider-google/pull/17929))
* privateca: added `subject_key_id` field to `google_privateca_certificate` and `google_privateca_certificate_authority` resources ([#17923](https://github.com/hashicorp/terraform-provider-google/pull/17923))
* secretmanager: added `version_destroy_ttl` field to `google_secret_manager_secret` resource ([#17888](https://github.com/hashicorp/terraform-provider-google/pull/17888))

BUG FIXES:
* appengine: added suppression for a diff in `google_app_engine_standard_app_version.automatic_scaling` when the block is unset in configuration ([#17905](https://github.com/hashicorp/terraform-provider-google/pull/17905))

> * sql: fixed issues with updating the `enable_google_ml_integration` field in `google_sql_database_instance` resource ([#17878](https://github.com/hashicorp/terraform-provider-google/pull/17878))

Author: Sorry about this - I've updated main with the 5.27.0 release notes, but after the release branch was created
